.github/workflows/build-image-job.yaml

```diff
@@ -0,0 +1,118 @@
+name: Build docker image
+permissions:
+  actions: read
+  checks: read
+  contents: write
+  deployments: read
+  discussions: read
+  id-token: write
+  issues: read
+  packages: write
+  pages: read
+  pull-requests: read
+  repository-projects: read
+  statuses: read
+  security-events: read
+
+on:
+  workflow_call:
+    inputs:
+      dockerfile:
+        description: 'Path to the Dockerfile'
+        required: true
+        type: string
+      registry:
+        description: 'Docker registry to push to'
+        required: false
+        default: 'docker.io'
+        type: string
+      image:
+        description: 'Name of the image'
+        required: true
+        type: string
+      image_tag:
+        description: 'Tag of the image'
+        required: true
+        type: string
+      push:
+        description: 'Push the image to the registry'
+        required: false
+        type: boolean
+        default: true
+      platforms:
+        description: 'Platforms to build for'
+        required: false
+        default: 'linux/amd64'
+        type: string
+    secrets:
+      username: 
+        description: 'Docker registry username'
+        required: false
+      password:
+        description: 'Docker registry token (password)'
+        required: false
+
+jobs:
+  docker:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+    steps:
+      - name: Print inputs
+        run: |
+          echo "dockerfile: ${{ inputs.dockerfile }}"
+          echo "registry: ${{ inputs.registry }}"
+          echo "image: ${{ inputs.image }}"
+          echo "image_tag: ${{ inputs.image_tag }}"
+          echo "push: ${{ inputs.push }}"
+          echo "platforms: ${{ inputs.platforms }}"
+          
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
+
+      - name: Set up Docker Buildx
+        id: buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Docker registry login
+        if: ${{ inputs.push }}
+        uses: docker/login-action@v3
+        with:
+          registry: ${{ inputs.registry }}
+          username: ${{ secrets.username }}
+          password: ${{ secrets.password }}
+
+      # https://github.com/orgs/community/discussions/27086
```

Comment:
This workaround is only for ghcr, so it should look more like:
```
      # https://github.com/orgs/community/discussions/27086
      - name: Set lower case image name
        if ${{ inputs.registry }} == "ghcr.io"
        run: |
          echo "IMAGE=${IMAGE,,}" >>${GITHUB_ENV}
        env:
          IMAGE: ${{ inputs.registry }}/${{ inputs.image }}
```

Reply:
Also for Docker it's needed

Reply:
https://docs.docker.com/docker-hub/repos/create/#:~:text=The%20repository%20name%20needs%20to,)%2C%20and%20underscores%20(%20_%20)